includes/class-cli.php

```diff
@@ -21,17 +23,289 @@ class Cli extends \WP_CLI_Command {
+	 * - Providing guidance on next steps
+	 *
+	 * Status can be:
+	 * - NOT STARTED: Process hasn't been initiated
+	 * - IN PROGRESS: Delete activities are still being processed
+	 * - COMPLETED: All Delete activities have been sent
 	 *
 	 * @return void
 	 */
-	public function self_destruct( $args, $assoc_args ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		\WP_CLI::warning( 'Self-Destructing is not implemented yet.' );
+	private function show_self_destruct_status() {
+		// Only proceed if self-destruct is active.
+		if ( ! \get_option( 'activitypub_self_destruct', false ) ) {
+			\WP_CLI::line( \WP_CLI::colorize( '%C❌ Status: NOT STARTED%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%CThe self-destruct process has not been initiated.%n' ) );
+			\WP_CLI::line( '' );
+			\WP_CLI::line( \WP_CLI::colorize( '%C💡 To start the process, run:%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%C   wp activitypub self_destruct%n' ) );
+			\WP_CLI::line( '' );
+			return;
+		}
+
+		\WP_CLI::line( \WP_CLI::colorize( '%B🔍 Self-Destruct Status Check%n' ) );
+		\WP_CLI::line( '' );
+
+		// Check if there are any more pending Delete activities for self-destruct.
+		$pending_deletes = \get_posts(
+			array(
+				'post_type'      => Outbox::POST_TYPE,
+				'post_status'    => 'pending',
+				'posts_per_page' => -1,
+				'fields'         => 'ids',
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'     => array(
+					array(
+						'key'     => '_activitypub_activity_type',
+						'value'   => 'Delete',
+						'compare' => '=',
+					),
+				),
+			)
+		);
+
+		// Get count of pending Delete activities.
+		$pending_count = count( $pending_deletes );
+
+		// If no more pending Delete activities, self-destruct is complete.
+		if ( 0 === $pending_count ) {
+			\WP_CLI::line( \WP_CLI::colorize( '%G✅ Status: COMPLETED%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%GYour blog has been successfully removed from the Fediverse.%n' ) );
+			\WP_CLI::line( '' );
+			\WP_CLI::line( \WP_CLI::colorize( '%Y📋 What happened:%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%Y• Delete activities were sent to all followers%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%Y• Your blog is no longer discoverable on ActivityPub networks%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%Y• The self-destruct process has finished%n' ) );
+		} else {
+			\WP_CLI::line( \WP_CLI::colorize( '%Y⏳ Status: IN PROGRESS%n' ) );
+			\WP_CLI::line( \WP_CLI::colorize( '%YThe self-destruct process is currently running.%n' ) );
+			\WP_CLI::line( '' );
+
+			if ( $pending_count > 0 ) {
+				\WP_CLI::line( \WP_CLI::colorize( "%Y📊 Progress: {$pending_count} Delete Activities still pending%n" ) );
+			} else {
+				\WP_CLI::line( \WP_CLI::colorize( '%Y📊 Progress: All Delete Activities have been processed%n' ) );
+				\WP_CLI::line( \WP_CLI::colorize( '%Y⚠️  Completion notification may be pending%n' ) );
+			}
```

Comment:
```suggestion
			\WP_CLI::line( \WP_CLI::colorize( "%Y📊 Progress: {$pending_count} Delete Activities still pending%n" ) );
```
If `$pending_count` is not more than 0, the `if` clause on line 284 will catch it.